docs/Recipes/faking-http-client.md

```diff
@@ -0,0 +1,98 @@
+
+Let's assume that you want to create a fake `HttpClient` so you can dictate the
+behavior of the
+[GetAsync(String)](https://learn.microsoft.com/en-us/dotnet/api/system.net.http.httpclient.getasync?view=net-7.0#system-net-http-httpclient-getasync(system-string))
+method. This seems like it would be a
+straightforward task, but it's complicated by the design of `HttpClient`, which
+is not faking-friendly.
+
+## A working Fake
+
+First off, let's look at the declaration of `GetAsync`:
+
+```csharp
+public Task<HttpResponseMessage>
+    GetAsync([StringSyntax(StringSyntaxAttribute.Uri)] string? requestUri)
+```
+
+This method is neither virtual nor abstract, and so [can't be overridden by
+FakeItEasy](../what-can-be-faked.md#what-members-can-be-overridden).
+
+As a workaround, we can look at the [definition of
+GetAsync](https://github.com/dotnet/runtime/blob/ab5e28c1cab305450897749daa7393bef30d7505/src/libraries/System.Net.Http/src/System/Net/Http/HttpClient.cs#L363-L364)
+and see that the method eventually ends up calling
+[HttpMessageHandler.SendAsync(HttpRequestMessage,
+CancellationToken)](https://learn.microsoft.com/en-us/dotnet/api/system.net.http.httpmessagehandler.sendasync?view=net-7.0#system-net-http-httpmessagehandler-sendasync(system-net-http-httprequestmessage-system-threading-cancellationtoken))
+on an `HttpMessageHandler` that can be supplied via the `HttpClient`
+constructor.
+
+`HttpMessageHandler.SendAsync` is protected, which makes it
+less convenient to override than a public method. We need to specify the call by
+name, and to give FakeItEasy a hint about the return type, as described in
+[Specifying a call to any method or
+property](../specifying-a-call-to-configure.md#specifying-a-call-to-any-method-or-property).
+
+With this knowledge, we can write a passing test:
+
+??? note "This is a simplified example"

```

Comment:
Does `??? node` do something special that I'm not aware of? Or is it just a placeholder for something that will actually render as a note?

Reply:
It does something you're not aware of! I thought you'd like it.

![image](https://user-images.githubusercontent.com/3275797/225489817-ccf121de-64ab-4b17-badb-cd4d5a616baf.png)

![image](https://user-images.githubusercontent.com/3275797/225489850-2145d38b-0947-4bcb-8f6d-a95948713c21.png)

Reply:
I do like it!
It's specific to mkdocs, right?

Reply:
Specifically it's a [PyMdown Extention](https://facelessuser.github.io/pymdown-extensions/extensions/details/).